### PR TITLE
enable view assessments on group dashboard

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
group view

screen shot 2018-04-30 at 4 53 15 pm

group dashboard and select cards

screen shot 2018-04-30 at 4 53 35 pm

after clicking view assessment 

screen shot 2018-04-30 at 4 53 42 pm

after going to "Most Recent"

screen shot 2018-04-30 at 4 53 47 pm

after clicking "Select Assessment" after viewing "Most Recent"

screen shot 2018-04-30 at 4 53 53 pm

### CODE REVIEW

File: webapp/src/main/webapp/src/app/assessments/assessments.component.ts [hunk 1]
```diff
@@ -35,10 +35,15 @@ export class AssessmentsComponent {
   @Input()
   set assessmentExams(value: AssessmentExam[]) {
     this._assessmentExams = value;
-    this.showOnlyMostRecent = true;
-    this._hasInitialAssessment = (value != null && value.length != 0);
+    const { assessmentIds } = this.route.snapshot.params;
+    if (!assessmentIds) {
+      this.showOnlyMostRecent = true;
+    }
+    this._hasInitialAssessment = (this._assessmentExams != null && this._assessmentExams.length != 0);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: i think `Utils` has a method for this

File: webapp/src/main/webapp/src/app/groups/results/group-results.component.html [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 <page-heading>
   <ng-template #heading>
-    <h2>{{'groups.name' | translate}}</h2>
+    <button *ngIf="viewGroupDashboard" class="btn btn-default btn-sm mb-sm" (click)="viewDashboard()">
+      <i class="fa fa-chevron-left mr-xs"></i>{{ 'group-results.view-dashboard' | translate}}
+    </button>
+    <h2 class="row">{{'groups.name' | translate}}</h2>
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/groups/results/group-results.component.html]: row?

File: webapp/src/main/webapp/src/app/assessments/assessments.component.ts [hunk 3]
```diff
@@ -173,14 +181,33 @@ export class AssessmentsComponent {
...
+    const { assessmentIds } = this.route.snapshot.params;
+    if (!assessmentIds) {
+      this.showOnlyMostRecent = true;
+    } else {
+      this.assessmentProvider.getAvailableAssessments().subscribe((value) => {
+        const selectedAssessments = value.filter((assessment) => assessmentIds.split(',').indexOf(assessment.id.toString()) >= 0);
+        selectedAssessments.forEach((assessment) => {
+          assessment.selected = true;
+          this.loadAssessmentExam(assessment);
+          this._preselectedAssessments = this._preselectedAssessments.concat(assessment);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: what does selected vs preselected mean?

File: webapp/src/main/webapp/src/app/groups/results/group-results.component.html (hunk 2 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/groups/results/group-results.component.html]: it looks like that is needed to get the "View Dashboard" and the "My Groups" on separate lines

File: webapp/src/main/webapp/src/app/assessments/assessments.component.ts (hunk 3 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: these are all preselected (renamed in last commit). Preselected are assessments select from the group-dashboard and loaded on initial page viewing of `/group` when clicking "View Assessments".

File: webapp/src/main/webapp/src/app/groups/results/group-results.component.html (hunk 2 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/groups/results/group-results.component.html]: oh, it should actually be in its own <page-heading> element then

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/groups/results/group-results.component.html]: can you elaborate? I tried moving the h2 into a `<div class="row">` but that caused the `h2` to be misaligned

Comment (esotrope) [reply webapp/src/main/webapp/src/app/groups/results/group-results.component.html]: just realized i missed the `` ticks on my comment. I think we need two `<page-header>` components. so basically move your `<h2>` into a new `<page-header>` that comes after the first one

Comment (esotrope) [reply webapp/src/main/webapp/src/app/groups/results/group-results.component.html]: actually scratch that - ill reach you over slack

**Pull request merged**